crates/nargo/src/cli/prove_cmd.rs

```diff
@@ -151,7 +158,18 @@ pub fn compile_circuit_and_witness<P: AsRef<Path>>(
             _ => unreachable!(),
         }
 
-    Ok((compiled_program, solved_witness))
+    Ok(solved_witness)
+}
+
+pub fn compile_circuit<P: AsRef<Path>>(
+    program_dir: P,
+    show_ssa: bool,
+) -> Result<noirc_driver::CompiledProgram, CliError> {
+    let driver = Resolver::resolve_root_config(program_dir.as_ref())?;
+    let backend = crate::backends::ConcreteBackend;
+    let compiled_program = driver.into_compiled_program(backend.np_language(), show_ssa);
+
+    Ok(compiled_program)
 }
```

Comment:
I'd move this into `compile_cmd.rs` as it's the more natural home for it.

Reply:
moving it, I agree